src/lib/viewers/image/ImageBaseViewer.js

```diff
@@ -434,6 +435,43 @@ class ImageBaseViewer extends BaseViewer {
         }
     }
 
+    /**
+     * Prints image using an an iframe.
+     *
+     * @return {void}
+     */
+    print() {
+        const browserName = Browser.getName();
+
+        /**
+         * Called async to ensure resource is loaded for print preview. Then removes listener to prevent
+         * multiple handlers.
+         *
+         * @return {void}
+         */
+        const defaultPrintHandler = () => {
+            if (browserName === BROWSERS.INTERNET_EXPLORER || browserName === BROWSERS.EDGE) {
+                this.printframe.contentWindow.document.execCommand('print', false, null);
+            } else {
+                this.printframe.contentWindow.print();
+            }
+
+            this.printframe.removeEventListener('load', defaultPrintHandler);
+            this.emit(VIEWER_EVENT.printSuccess);
+        };
+
+        this.printframe = openContentInsideIframe(this.imageEl.outerHTML);
+        this.printImages = this.printframe.contentDocument.querySelectorAll('img');
+
+        for (let i = 0; i < this.printImages.length; i += 1) {
```

Comment:
Minor, but if the index isn't needed, iterators are usually preferred:

```
this.printImages.forEach(printImage => {
    printImage.setAttribute('style', 'display: block; margin: 0 auto; width: 100%');
});
```

Reply:
IE11 doesn't support foreach from a NodeList, so instead of using something like `Array.prototype.slice.call(this.printImages).forEach(ele =>...` or different way of converting the nodelist to an array I just went with a simpler for loop.

Reply:
Ah, I didn't consider that. Sounds good!